services/ymax-planner/src/plan-deposit.ts

```diff
@@ -115,37 +116,35 @@ const computeWeightedTargets = (
   brand: Brand<'nat'>,
   current: Partial<Record<AssetPlaceRef, NatAmount>>,
   delta: bigint,
-  allocation: Record<PoolKey, number>,
+  allocation: TargetAllocation,
 ): Partial<Record<AssetPlaceRef, NatAmount>> => {
   const totalCurrentAmt = Object.keys(current).reduce(
     (acc, k) => (allocation[k] ? AmountMath.add(acc, current[k]) : acc),
     AmountMath.makeEmpty(brand),
   );
   const total = totalCurrentAmt.value + delta;
   assert(total >= 0n, X`total after delta must not be negative`);
-  const entries = Object.entries(allocation) as Array<
-    [PoolKey, number | bigint]
-  >;
+  const entries = Object.entries(allocation) as Array<[PoolKey, NatValue]>;
   assert(entries.length > 0, X`empty allocation`);
-  const SCALE_NUM = 1_000_000;
   const weights = entries.map(([k, w]) => {
-    const wNum = Number(w as any);
-    assert(Number.isFinite(wNum), X`allocation weight must be a number`);
-    const wScaled = BigInt(Math.round(wNum * SCALE_NUM));
-    return [k, wScaled] as const;
+    assert(
+      typeof w === 'bigint' && w >= 0n,
+      X`allocation weight must be a Nat`,
+    );
+    return [k, w] as const;
   });
   const sumW = weights.reduce<bigint>((acc, [, w]) => acc + w, 0n);
   assert(sumW > 0n, X`allocation weights must sum > 0`);
   const draft: Partial<Record<AssetPlaceRef, NatAmount>> = {};
   let assigned = 0n;
   let maxKey = entries[0][0];
-  let maxW = -1n as unknown as bigint;
+  let maxW = -1n as bigint;
```

Comment:
This still seems redundant.
```suggestion
  let maxW = -1n;
```